src/vivarium/framework/values.py

```diff
@@ -227,16 +227,43 @@ def __call__(self, *args, skip_post_processor=False, **kwargs):
         """
         return self._call(*args, skip_post_processor=skip_post_processor, **kwargs)
 
-    def _call(self, *args, skip_post_processor=False, **kwargs):
+    def _call(self, *args, skip_post_processor=False, use_cache=True, **kwargs):
         if not self.source:
             raise DynamicValueError(
                 f"The dynamic value pipeline for {self.name} has no source. This likely means "
                 f"you are attempting to modify a value that hasn't been created."
             )
 
-        value = self.source(*args, **kwargs)
-        for mutator in self.mutators:
-            value = self.combiner(value, mutator, *args, **kwargs)
+        # if only one kwarg was passed in and it's an index, treat it as the arg
+        if (len(args) == 0) and (len(kwargs) == 1) and (isinstance(kwargs["index"], pd.Index)):
+            args = [kwargs["index"]]
+            kwargs = {}
+
+        # if we only have one arg and it's an index
+        if (len(args) == 1) and (isinstance(args[0], pd.Index)) and (kwargs == {}) and use_cache:
+            if self.name in self.manager.call_history:
+                requested_idx = args[0]
+                previously_called_idx = self.manager.call_history[self.name].index
+
+                newly_requested = requested_idx.difference(previously_called_idx)
+            else:
+                self.manager.call_history[self.name] = pd.DataFrame()
+                newly_requested = args[0]
+            if newly_requested.any():
+                newly_requested = newly_requested.unique()
+                value = self.source(newly_requested)
+                for mutator in self.mutators:
+                    value = self.combiner(value, mutator, newly_requested)
+
+                self.manager.call_history[self.name] = self.manager.call_history[self.name].append(value)
```

Comment:
What I am curious about is we update previously_called_idx each time we request a new index.  As someone that does not have a great understand of caching, does this mean if we request a subset of that index we would still have a cache for that or would we need to make that call once and then have the cache?